content/blog/2022-05-06-end-to-end-computer-vision-api-part-3-remote-exp-ci-cd.md

```diff
@@ -0,0 +1,293 @@
+time and assess the quality of the returned segmentation masks.
+
+#### 3. [Workflow for deploying to the production environment](https://github.com/iterative/magnetic-tiles-defect/blob/main/.github/workflows/3-deploy.yaml)
+
+![Workflow for deploying to the production environment](/uploads/images/2022-05-06/workflow_prod.png '=400')
+_Workflow for deploying to the production environment_
+
+If there are no issues and we are confident in the quality of the new model, we
+can merge the development branch into the main branch of our repository. Again,
+this triggers the third CI/CD workflow that deploys the code from the main
+branch to the production API. This looks identical to the deployment into the
+development environment, except now the deployment endpoint will be
+
+<https://demo-api-mag-tiles-prod.herokuapp.com/analyze> (note `-prod`) .
+
+## Summary
+
+In this series of posts (see
+[part 1](https://dvc.org/blog/end-to-end-computer-vision-api-part-1-data-versioning-and-ml-pipelines)
+and
+[part 2](https://dvc.org/blog/end-to-end-computer-vision-api-part-2-local-experiments)),
+we described how we addressed the problem of building a Computer Vision Web API
+for defect detection. We’ve chosen this approach because it addresses the common
+challenges that are shared across many CV projects: how to version datasets that
+consist of a large number of small- to medium-sized files; how to avoid
+triggering long-running stages of an ML pipeline when it’s not needed for
+reproducibility; how to run model training jobs on the cloud infrastructure
+without having to provision and manage everything yourself; and, finally, how to
+track progress in key metrics when you run many ML experiments.
+
+We've talked about the following:
+
+- Common difficulties when building Computer Vision Web API for defect detection
+  ([link](https://dvc.org/blog/end-to-end-computer-vision-api-part-1-data-versioning-and-ml-pipelines#introduction))
+- Pros and cons of exploratory work in Jupyter Notebooks
+  ([link](https://dvc.org/blog/end-to-end-computer-vision-api-part-1-data-versioning-and-ml-pipelines#proof-of-concept-in-jupyter-notebooks))
+- Versioning data in remote storage with DVC
+  ([link](https://dvc.org/blog/end-to-end-computer-vision-api-part-1-data-versioning-and-ml-pipelines#data-versioning))
+- Moving and refactoring the code from Jupyter Notebooks into DVC pipeline
+  stages
+  ([link](https://dvc.org/blog/end-to-end-computer-vision-api-part-1-data-versioning-and-ml-pipelines#refactoring-jupyter-code-into-an-ml-pipeline))
+- Experiment management and versioning
+  ([link](https://dvc.org/blog/end-to-end-computer-vision-api-part-2-local-experiments#experiment-management))
+- Visualization of experiments and collaboration in Iterative Studio
+  ([link](https://dvc.org/blog/end-to-end-computer-vision-api-part-2-local-experiments#collaboration-and-reporting-with-iterative-studio))
+- Remote experiments, CI/CD, and production deployment (this post)
```

Comment:
Missing link

Reply:
Oh, that’s on purpose. I didn’t want to link from the post to itself.

Reply:
Oh, gotcha.